### ISSUE TITLE
Exception in Rider 2022.3

### ISSUE THREAD

**Issue by lstatie** [izhangzhihao/intellij-rainbow-brackets#2529]
I am using Rider 2022.3 EAP 3.
Build #RD-223.6160.184, built on October 8, 2022

Licensed to JetBrains Rider EAP user: Lorenzo Statie
Expiration date: November 7, 2022

Runtime version: 17.0.4.1+1-b629.2 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

.NET 6.0.8 (Server GC)

Rainbow Brackets (6.26) throws an exception in Rider.

**_Stacktrace:_**
```
com.intellij.diagnostic.PluginException: Cannot create extension (class=com.github.izhangzhihao.rainbow.brackets.visitor.CSharpRainbowVisitor) [Plugin: izhangzhihao.rainbow.brackets]
	at com.intellij.serviceContainer.ComponentManagerImpl.createError(ComponentManagerImpl.kt:1058)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.doCreateInstance(XmlExtensionAdapter.kt:70)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.kt:31)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:473)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:421)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensions(ExtensionPointImpl.java:264)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensions(ExtensionPointName.java:91)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.cloneHighlightVisitors(GeneralHighlightingPass.java:108)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.getHighlightVisitors(GeneralHighlightingPass.java:146)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:166)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:97)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:57)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:405)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1086)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:397)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:666)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:622)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:590)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:396)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:372)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:370)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.intellij.diagnostic.PluginException: Cannot create class com.github.izhangzhihao.rainbow.brackets.visitor.CSharpRainbowVisitor (classloader=PluginClassLoader(plugin=PluginDescriptor(name=Rainbow Brackets, id=izhangzhihao.rainbow.brackets, descriptorPath=plugin.xml, path=~\AppData\Roaming\JetBrains\Rider2022.3\plugins\intellij-rainbow-brackets, version=6.26, package=null, isBundled=false), packagePrefix=null, instanceId=142, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:996)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.instantiateClass(XmlExtensionAdapter.kt:79)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter$SimpleConstructorInjectionAdapter.instantiateClass(XmlExtensionAdapter.kt:90)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.doCreateInstance(XmlExtensionAdapter.kt:50)
	... 29 more
Caused by: java.lang.NoClassDefFoundError: com/jetbrains/rider/ideaInterop/fileTypes/csharp/lexer/CSharpTokenType
	at com.github.izhangzhihao.rainbow.brackets.visitor.CSharpRainbowVisitor.<clinit>(CSharpRainbowVisitor.kt:53)
	at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
	at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:520)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:981)
	... 32 more
Caused by: java.lang.ClassNotFoundException: com.jetbrains.rider.ideaInterop.fileTypes.csharp.lexer.CSharpTokenType PluginClassLoader(plugin=PluginDescriptor(name=Rainbow Brackets, id=izhangzhihao.rainbow.brackets, descriptorPath=plugin.xml, path=~\AppData\Roaming\JetBrains\Rider2022.3\plugins\intellij-rainbow-brackets, version=6.26, package=null, isBundled=false), packagePrefix=null, instanceId=142, state=active)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:215)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 36 more
```

**Comment by TraceLD** [comment]
I'm getting the same issue in Rider 2022.3 EAP 3

**Comment by izhangzhihao** [comment]
Will be fixed in the next release(the end of this month)

**Comment by izhangzhihao** [comment]
Hi, the update is already in the marketplace, please download and enjoy it: https://plugins.jetbrains.com/plugin/10080-rainbow-brackets/versions/stable/244619

By the way, I create a new Google Search plugin, please try and give me feedback: https://plugins.jetbrains.com/plugin/20230-google-search/

**Comment by fluffynuts** [comment]
Since you locked the original thread, but mentioned this, so it's now over here:|

1. 
> YOU ALWAYS HAVE A CHOICE

I don't, because my choice is to update Rider _and_ rainbow brackets or _neither_. Since I already have an opensource license from the kind people at JetBrains for all of Toolbox, and since Rider still has outstanding stuff which needs to get fixed / implemented, I can't not update that.

2.
> are you scaring me？ You have used this plugin for free for so many years, without a line of code contribution, without a dollar donation, do you have a little bit of respect for the authors of this plugin?

I don't mind a one-off donation. I would have gladly understood that. I don't place any extra burden on you as a user of the plugin that I was happy with from the very first version I ever used, years ago. I don't raise issues. I cost you literally nothing. I went to have a look at the payment options and the only thing I could see there was a recurring payment, which I feel is unfair because I literally only wanted the single feature that was always free: rainbow brackets. Nothing else.

> Open source code is always open source, open source code is always free, this is my principle. But since 2022.3 the original open-source code is no longer available, mind you.

So, open-source is open-source until you pull the rug out from under people. Got it.

3.
> As long as you have contributed a line of code or donated a dollar before, you will definitely get a permanent free license, I promise. but you didn't

What code should I have contributed to a plugin which did exactly what it promised on the box from the first day I ever used it? It provided per-level bracket coloring. Perfect! What could I possibly add to that? I'm sorry I didn't donate before - I must have missed the wording to do so. I did see upgrade notices every time, but never really took the time to read them - I don't know about you, but when I open an IDE, it's to get code done, so I kinda focus on what I came there for.

4. It looks like Rainbow Brackets is still free-to-use for other languages. I'm interested as to why C# specifically was targeted for this? It feels like some kind of grudge or an assumption that people working on dotnet just have money to chuck about? I have quite a few opensource dotnet projects, millions of downloads on nuget.org, all for stuff that I don't charge anyone a cent for. But apparently, you've decided that since I didn't contribute directly to _your_ work, I don't matter.

5. Locking off the thread at https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2565 after you've added your bit, not allowing a person to discuss, is, at best, rude. I could think of other terms for it.
